rhizome/lib/vm_setup.rb

```diff
@@ -255,6 +297,34 @@ def install_systemd_unit(max_vcpus, cpu_topology, mem_gib)
 ReadOnlyPaths=/
 DNSMASQ_SERVICE
 
+    cpu = r("ls -1 /vm/").split.size
+
+    vp.write_spdk_service <<SPDK_SERVICE
+[Unit]
+Description=Block Storage Service
+
+[Service]
+Type=simple
+Environment="XDG_RUNTIME_DIR=#{vp.home("")}"
+ExecStart=/opt/spdk/bin/vhost -S #{vp.home("")} \
+--huge-dir #{vp.q_hugepages} \
+--iova-mode va \
+--rpc-socket #{vp.q_spdk_sock} \
+--cpumask [#{cpu}] \
```

Comment:
We should probably always allocate core zero...but the reason for core zero is because it generally has the same NUMA node as PCI devices. Really we care about NUMA here, but by sticking to zero we probably satisfy our needs in most cases.

Reply:
note core zero may be two threads, you may want to consult vmhost cpu topology data to do the job.

Reply:
done